### PR TITLE
Docs - missed out date time functions

### PR DESCRIPTION
**Pull request by bharatnc**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Documentation (changelog entry is not required)

Detailed description / Documentation draft:

Documents missed date time functions.

Related to: #16049

### CODE REVIEW

File: docs/en/sql-reference/functions/date-time-functions.md [hunk 1]
```diff
@@ -435,6 +435,166 @@ Result:
...
+    - `second`
+    - `minute`
+    - `hour`
+    - `day`
+    - `week`
+    - `month`
+    - `quarter`
+    - `year`
+-   `value` - Amount of the specified unit of time.    
+-   `date` — Date or Date with time - [Date](../../sql-reference/data-types/date.md) or [DateTime](../../sql-reference/data-types/datetime.md).
```
Comment (vdimir) [inline docs/en/sql-reference/functions/date-time-functions.md:464]:
```suggestion
-   `date` — [Date](../../sql-reference/data-types/date.md) or [DateTime](../../sql-reference/data-types/datetime.md).
```

File: docs/en/sql-reference/functions/date-time-functions.md [hunk 2]
```diff
@@ -435,6 +435,166 @@ Result:
...
+select date_add(YEAR, 3, toDate('2018-01-01'));
+```
+
+```text
+┌─plus(toDate('2018-01-01'), toIntervalYear(3))─┐
+│                                    2021-01-01 │
+└───────────────────────────────────────────────┘
+```
+
+## date\_diff {#date_diff}
```
Comment (vdimir) [inline docs/en/sql-reference/functions/date-time-functions.md:479]: `dateDiff` already documented in this file https://github.com/ClickHouse/ClickHouse/blob/f7dbdc623cc4122f7b03f621e2c3f4c4d745b74f/docs/en/sql-reference/functions/date-time-functions.md#L553

You can add some information, e.g. aliases

File: docs/en/sql-reference/functions/date-time-functions.md [hunk 3]
```diff
@@ -435,6 +435,166 @@ Result:
...
+
+``` sql
+date_add(unit, value, date)
+```
+
+Aliases: `dateAdd`, `DATE_ADD`. 
+
+**Parameters**
+
+-   `unit` - The unit of time - [String](../syntax.md#syntax-string-literal).
```
Comment (den-crane) [inline docs/en/sql-reference/functions/date-time-functions.md:452]: only it's not String

'second' -- String
second -- Literal ?

File: docs/en/sql-reference/functions/date-time-functions.md [hunk 4]
```diff
@@ -435,6 +435,166 @@ Result:
 
 -   [toStartOfInterval](#tostartofintervaltime-or-data-interval-x-unit-time-zone)
 
+## date\_add {#date_add}
```
Comment (den-crane) [inline docs/en/sql-reference/functions/date-time-functions.md:438]: But how do these functions work?
What will happen if we will add one month to 31th January?

File: docs/en/sql-reference/functions/date-time-functions.md (hunk 3 above)
Comment (bharatnc) [reply docs/en/sql-reference/functions/date-time-functions.md]: yes, it's a `string literal`. Made the change.

File: docs/en/sql-reference/functions/date-time-functions.md (hunk 2 above)
Comment (bharatnc) [reply docs/en/sql-reference/functions/date-time-functions.md]: Right, good catch. Preserved the older one and added some additional info.

File: docs/en/sql-reference/functions/date-time-functions.md (hunk 3 above)
Comment (bharatnc) [reply docs/en/sql-reference/functions/date-time-functions.md]: I think it's the other way around:
'second' - string literal: https://github.com/ClickHouse/ClickHouse/blob/master/docs/en/sql-reference/syntax.md#string-syntax-string-literal. Example: `HOUR` `MINUTE` etc.
second - string: https://github.com/ClickHouse/ClickHouse/blob/master/docs/en/sql-reference/data-types/string.md#string-string. In this case `HOUR`, `MINUTE` etc.

For all these examples, it should actually be String.

File: docs/en/sql-reference/functions/date-time-functions.md (hunk 4 above)
Comment (bharatnc) [reply docs/en/sql-reference/functions/date-time-functions.md:438]: I think these functions were written to make them compatible with MySQL functions. For example https://www.w3schools.com/sql/func_mysql_date_add.asp. 
So I assume that they are expected to work how they work with MySQL ?
To answer your question adding 1 month to Jan 31st:
```
select DATE_ADD(MONTH, 1, toDateTime('2021-01-31'));

┌─plus(toDateTime('2021-01-31'), toIntervalMonth(1))─┐
│                                2021-02-28 00:00:00 │
└────────────────────────────────────────────────────┘
```

Comment (den-crane) [reply docs/en/sql-reference/functions/date-time-functions.md:438]: https://www.w3schools.com/sql/func_mysql_date_add.asp
The type of interval to add. Can be one of the following      values:            MICROSECOND  

probably we can use the same sentence `The type of interval `

Comment (bharatnc) [reply docs/en/sql-reference/functions/date-time-functions.md:438]: Sure, reworded the def of `unit` to use `The type of interval ....` wherever relevant.

File: docs/en/sql-reference/functions/date-time-functions.md [hunk 5]
```diff
@@ -435,6 +435,161 @@ Result:
...
+
+-   `startdate` — The first time value to compare. [Date](../../sql-reference/data-types/date.md) or [DateTime](../../sql-reference/data-types/datetime.md).
+
+-   `enddate` — The second time value to compare. [Date](../../sql-reference/data-types/date.md) or [DateTime](../../sql-reference/data-types/datetime.md).
+
+-   `timezone` — Optional parameter. If specified, it is applied to both `startdate` and `enddate`. If not specified, timezones of `startdate` and `enddate` are used. If they are not the same, the result is unspecified.
+
+**Returned value**
+
+Difference between `startdate` and `enddate` expressed in `unit`.
```
Comment (vdimir) [inline docs/en/sql-reference/functions/date-time-functions.md:501]:
```suggestion
Difference between `enddate` and `startdate` expressed in `unit`.
```

File: docs/en/sql-reference/functions/date-time-functions.md [hunk 6]
```diff
@@ -435,6 +435,161 @@ Result:
...
+
+**Parameters**
+
+-   `unit` — The type of interval to add. [String](../../sql-reference/data-types/string.md).
+
+        Supported values: second, minute, hour, day, week, month, quarter, year.
+
+-   `startdate` — The first time value to compare. [Date](../../sql-reference/data-types/date.md) or [DateTime](../../sql-reference/data-types/datetime.md).
+
+-   `enddate` — The second time value to compare. [Date](../../sql-reference/data-types/date.md) or [DateTime](../../sql-reference/data-types/datetime.md).
```
Comment (vdimir) [inline docs/en/sql-reference/functions/date-time-functions.md:495]:
```suggestion
-   `startdate` — The first time value to subtract (the subtrahend). [Date](../../sql-reference/data-types/date.md) or [DateTime](../../sql-reference/data-types/datetime.md).

-   `enddate` — The second time value to subtract from (the minuend). [Date](../../sql-reference/data-types/date.md) or [DateTime](../../sql-reference/data-types/datetime.md).
```

File: docs/en/sql-reference/functions/date-time-functions.md [hunk 7]
```diff
@@ -435,6 +435,161 @@ Result:
...
+
+``` sql
+date_diff('unit', startdate, enddate, [timezone])
+```
+
+Aliases: `dateDiff`, `DATE_DIFF`. 
+
+**Parameters**
+
+-   `unit` — The type of interval to add. [String](../../sql-reference/data-types/string.md).
```
Comment (vdimir) [inline docs/en/sql-reference/functions/date-time-functions.md:489]:
```suggestion
-   `unit` — The type of interval for result [String](../../sql-reference/data-types/string.md).
```

File: docs/en/sql-reference/functions/date-time-functions.md [hunk 8]
```diff
@@ -380,7 +380,7 @@ Alias: `dateTrunc`.
 
 **Parameters**
 
--   `unit` — Part of date. [String](../syntax.md#syntax-string-literal).
+-   `unit` — The type of interval to add. [String](../../sql-reference/data-types/string.md).
```
Comment (vdimir) [inline docs/en/sql-reference/functions/date-time-functions.md:383]:
```suggestion
-   `unit` — The type of interval to truncate the result. [String](../../sql-reference/data-types/string.md).
```

**Review by vdimir — APPROVED** [review]
Thank you for your changes!
I'll only made minor changes (that forgotten to be changed in copypaste I guess).

**Pull request merged**